### DISCUSSION TITLE
problem installing pip on mac

### DISCUSSION THREAD

**Discussion by Elizabethbest** [bitcraze/discussions#620] [Q&A]
Hi there
im trying to install python (3.11) and the  cf client on my macbook (M1) .

I am a total beginner in all this. But have followed the instruction's online to install but keep getting stuck at one point. it comes with this error: 
1 error generated.
      ************************************************
      ************************************************
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyzmq
Failed to build pyzmq
ERROR: Could not build wheels for pyzmq, which is required to install pyproject.toml-based projects
MacBook-Pro-2:~ best$ 

my goal is to make a drone act on stage using min 4 crazyflie up to 10 crazyflie. 
Using Collmot program - skybrush and skybrush studio for blender.
but so far I haven't been able to install the cf client. 
Python seems to be installed however when i try to open it , it opena a preferences tab, and dont know what i should be clicking on. any help is much appreciated

**Comment by hmllr** [comment]
Which command fails exactly? It looks like you might need to upgrade pip (`python -m pip install --upgrade pip` and then 
`python -m pip install --upgrade setuptools`). 
Let us know if this helps,
Hanna

**Reply by Elizabethbest** [reply]
hi there,
so i did what you recommeded and this is what i got.... not found...
last login: Sun Mar 12 16:11:57 on ttys001

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
MacBook-Pro-2:~ best$ "python -m pip install --upgrade pip"
-bash: python -m pip install --upgrade pip: command not found
MacBook-Pro-2:~ best$ "python -m pip install --upgrade setuptools"
-bash: python -m pip install --upgrade setuptools: command not found
MacBook-Pro-2:~ best$

**Reply by hmllr** [reply]
Ah, sorry, this might have been unclear in my answer - you are not supposed to enter the " signs as well, they were just there to mark the begin and end. I updated it above with proper formatting. It might also be that you have to use `python3` instead of `python`.

Another solution would be to use the virtual machine, where everything is already installed: https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/#inst-virtualmachine

Hope this helps;
Hanna

**Reply by Elizabethbest** [reply]
Hi Hanna 
ahh ok, will try again.
So i have already installed python 3.11
I wasn't sure what the virtual machine was, but sounds interesting, i will have a look now on it

**Reply by hmllr** [reply]
just to clarify, this is not about installing python, this is about installing pip, which is a package installer for python.

**Reply by Elizabethbest** [reply]
yes. Python seems to be installed ok. It's just pip.

**Reply by Elizabethbest** [reply]
> Hi Hanna ahh ok, will try again. So i have already installed python 3.11 I wasn't sure what the virtual machine was, but sounds interesting, i will have a look now on it

so i tried the VM but its not possible on my macbook , i have M1.  Also still coming up with same results, when I tried to install pip

**Reply by hmllr** [reply]
yes, the virtual machine on your mac might be a bit tricky, we haven't tried this ourselves actually.

for installing the client natively - even though it is not the problem right now, you'll need to switch to python3.10, as the client doesn't support 3.11 yet (we are looking into it). So you should try to follow the instructions here: https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/installation/install/#mac but exchange python3 with python3.10

Hope this helps.

**Reply by Elizabethbest** [reply]
ok. so i have installed python 3.10. looks good. 
and got to here:# Installing brew for x86_64, it will be installed in /usr/local by default
arch --x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
# Now we have to use brew and then python from /usr/local ...
arch --x86_64 brew install python@3.9 libusb
# The arch command is not required anymore since everything brew installed are x86 executables

i copy and pasted the commands , but second command gave me this back:
MacBook-Pro-2:~ best$ arch
×86_64 brew install python@3.9 libusb
fac Running
"brew update
-auto-update
jpg
==> Homebrew is run entirely by unpaid volunteers. Please consider donating:
https://github.com/Homebrew/brew#donations
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> New Formulae
adr-viewer
ig_parttime
benerator
libdvbcsa
phoneinfoga
portal
smartdns
You have 3 outdated formulae
installed.
You can upgrade them with brew upgrade
or list them with brew outdated.
Warning: libusb1.0.26 is already installed and up-to-date.
To reinstall 1.0.26, run:
brew reinstall libusb
vert
Error: Cannot install under Rosetta 2 in ARM default prefix (/opt/homebrew)!
ller
To rerun under ARM use:
arch
-arm64 brew install
To install under ×86_64, install Homebrew into /usr/local.

so i need to type this in??
install Homebrew into /usr/local.

im not sure, any help?

**Reply by krichardsson** [reply]
Hi!

It seems as your problems (at this point) are mainly related to installing python on a M1 MAC. I think you should try to follow some tutorial on the internet to get some basic knowledge about your system and python before diving into the Crazyflie ecosystem. We're happy to help out with any problems related to our code and systems, but unfortunately we don't have the resources (or knowledge) to help you with the problems you have. Please don't hesitate to get back to us if you run into any problems that you think is related to our code.

Thanks!
